src/lib/datepicker/datepicker-input.ts

```diff
@@ -380,7 +378,8 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
     }
 
     this._elementRef.nativeElement.value =
-        value ? this._dateAdapter.format(value, this._dateFormats.display.dateInput) : '';
+        value && this._getValidDateOrNull(value) ?
```

Comment:
not necessary to check `value &&` since `_getValidDateOrNull` accepts `any`

Reply:
True, but the compiler will complain in the follow up line in `this._dateAdapter.format(value...` since format doesn't accept null.